### PR TITLE
Feature/add split docs

### PR DESCRIPTION
**Pull request by GeeH**
Split docs from main zf2-documentation and converted to markdown.

Needs review as this is more Jedi mind tricks that real scripting code.

Tables should now convert, and includes should get included.

### CODE REVIEW

**Comment by froschdesign** [comment]
@GeeH 
Do you use a script or software to convert the documentation?

**Comment by weierophinney** [comment]
@froschdesign — @GeeH is working on a script for converting the docs, and using the first few submissions to components to work out the kinks.

**Comment by froschdesign** [comment]
@weierophinney and @GeeH 
Do you know [Pandoc](http://pandoc.org)?

**Comment by weierophinney** [comment]
@froschdesign IIRC, @GeeH is using pandoc as part of the process. However, some elements do not translate, and others we want to translate to very specific markup.

File: doc/book/zend.barcode.creation.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+Barcode creation using Zend\\Barcode\\Barcode class
+===================================================
```
Comment (weierophinney) [inline doc/book/zend.barcode.creation.md]: Above should use the `#` style heading.

File: doc/book/zend.barcode.creation.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Using Zend\\Barcode\\Barcode::factory
+-------------------------------------
```
Comment (weierophinney) [inline doc/book/zend.barcode.creation.md]: Above should use the `##` style heading.

File: doc/book/zend.barcode.creation.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+object (required)
+- The name of the renderer (e.g., "image") (required)
+- Options to pass to the barcode object (an array or a [Traversable](http://php.net/traversable)
+object) (optional)
+- Options to pass to the renderer object (an array or a [Traversable](http://php.net/traversable)
+object) (optional)
+- Boolean to indicate whether or not to automatically render errors. If an exception occurs, the
+provided barcode object will be replaced with an Error representation (optional default `TRUE`)
+
+**Getting a Renderer with Zend\\Barcode\\Barcode::factory()**
```
Comment (weierophinney) [inline doc/book/zend.barcode.creation.md]: This should be a `###` style heading. My guess is we won't be able to automate this, though.

File: doc/book/zend.barcode.creation.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+// No required options
+$rendererOptions = array();
+$renderer = Barcode::factory(
+    'code39', 'image', $barcodeOptions, $rendererOptions
+);
+
+```
+
+**Using Zend\\Barcode\\Barcode::factory() with Zend\\Config\\Config objects**
```
Comment (weierophinney) [inline doc/book/zend.barcode.creation.md]: Same with this one.

File: doc/book/zend.barcode.objects.md [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+|getQuietZone()                     |Integer  |Return the size of the space needed before and after the barcode without any drawing                                   |
+|getInstructions()                  |Array    |Return drawing instructions as an array.                                                                               |
+|getHeight($recalculate = false)    |Integer  |Return the height of the barcode calculated after possible rotation                                                    |
+|getWidth($recalculate = false)     |Integer  |Return the width of the barcode calculated after possible rotation                                                     |
+|getOffsetTop($recalculate = false) |Integer  |Return the position of the top of the barcode calculated after possible rotation                                       |
+|getOffsetLeft($recalculate = false)|Integer  |Return the position of the left of the barcode calculated after possible rotation                                      |
+
+
+
+orphan
```
Comment (weierophinney) [inline doc/book/zend.barcode.objects.md]: What's this?

File: doc/book/zend.barcode.renderers.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+
+|Option   |Data Type|Default Value|Description                                                                                                      |
+|---------|---------|-------------|-----------------------------------------------------------------------------------------------------------------|
+|height   |Integer  |0            |Allow you to specify the height of the result image. If "0", the height will be calculated by the barcode object.|
+|width    |Integer  |0            |Allow you to specify the width of the result image. If "0", the width will be calculated by the barcode object.  |
+|imageType|String   |"png"        |Specify the image format. Can be "png", "jpeg", "jpg" or "gif".                                                  |
+
+ ..
+\_zend.barcode.renderers.pdf:
```
Comment (weierophinney) [inline doc/book/zend.barcode.renderers.md]: Odd — wonder why this wasn't stripped? (it's a named reference marker)

**Comment by weierophinney** [comment]
@GeeH Looks like there's a few more things to add to the rules:
- Conversion of heading styles.
- May want to look for odd, manual style headers (`**This is bolded to emulate a header**`) and convert them; the problem is determining the level. We may have to do these manually.
- It looks like the old style reference markers (e.g., `.. _zend.barcode.renderers.pdf:`) are not being stripped; we'll need to do some sort of regex for removing these.

Overall, though, looks really, really good!

File: doc/book/zend.barcode.objects.md (hunk 5 above)
Comment (GeeH) [inline doc/book/zend.barcode.objects.md]: Renders as a table check preview

Comment (weierophinney) [inline doc/book/zend.barcode.objects.md]: Bizarre.

Comment (GeeH) [inline doc/book/zend.barcode.objects.md]: Oh my bad, let me give this another go later no idea why the headings didn't work but I looked at this so long I was missing obvious stuff.

Comment (froschdesign) [inline doc/book/zend.barcode.objects.md]: `orphan`:  http://sphinx-doc.org/markup/misc.html#file-wide-metadata

**Comment by GeeH** [comment]
I'll tweak the script and find out why stuff that was working isn't working. Are you OK if we aim for 90% then ask people to contribute by manually tidying each repo?

File: doc/book/zend.barcode.objects.md (hunk 5 above)
Comment (weierophinney) [inline doc/book/zend.barcode.objects.md]: Okay, so those can be removed now.

**Comment by weierophinney** [comment]
> Are you OK if we aim for 90% then ask people to contribute by manually tidying each repo?

Absolutely.

**Comment by GeeH** [comment]
ping @weierophinney :beers:

File: doc/book/zend.barcode.objects.md [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+|withQuietZones    |Boolean                          |TRUE                 |Leave a quiet zone before and after the barcode                                                            |
+|drawText          |Boolean                          |TRUE                 |Set if the text is displayed below the barcode                                                             |
+|stretchText       |Boolean                          |FALSE                |Specify if the text is stretched all along the barcode                                                     |
+|withChecksum      |Boolean                          |FALSE                |Indicate whether or not the checksum is automatically added to the barcode                                 |
+|withChecksumInText|Boolean                          |FALSE                |Indicate whether or not the checksum is displayed in the textual representation                            |
+|text              |String                           |NULL                 |The text to represent as a barcode                                                                         |
+
+ ..
+\_zend.barcode.barcode.common.options.barcodefont:
```
Comment (weierophinney) [inline doc/book/zend.barcode.objects.md]: This is another of the reference markers. If they're coming in like this, we can likely strip them with a regex like the following:

```
/[^|\n]\.\.\s*\n\\_[a-z0-9._-]+:\s*\n/s
```

Use that as the match argument of a `preg_replace`, and make the replacement simply `"\n"`, and it should be good to go.

File: doc/book/zend.barcode.objects.md [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+- **Length:** variable (always even number of characters)
+
+Available options include:
+
+
+|Option        |Data Type|Default Value|Description                                               |
+|--------------|---------|-------------|----------------------------------------------------------|
+|withBearerBars|Boolean  |FALSE        |Draw a thick bar at the top and the bottom of the barcode.|
+
+ .. note:
```
Comment (weierophinney) [inline doc/book/zend.barcode.objects.md]: Cannot recall: didn't you find a way to convert notes to blockquotes or similar, @GeeH?

**Comment by weierophinney** [comment]
@GeeH Looks great! just two problems, one of which I think can be automated, the last one I think _is_ automated, but likely an edge case (the note — I say edge case, as I see that other notes _were_ formatted correctly, which makes me think there may be a slight difference in the markup for this particular one).

**Comment by weierophinney** [comment]
:+1: Will merge soon!

**Comment by KimPrince** [comment]
I'd like to help by manually tidying up the repos.

**Comment by weierophinney** [comment]
@KimPrince Not sure what you're proposing, so create a pull request or two on one or more repos to demonstrate. :)

**Pull request closed without merging**